backend/src/backend/main.py

```diff
@@ -97,25 +103,37 @@ async def add_book(book_payload: BookAddSchema):
                 kind=KIND_TEXT,
             )
 
-    title = await Title.objects.get_or_create(
-        ident=get_ident_from_name(book_payload.metadata["Name"])
-    )
-    await book.update(title=title)
+    for metadata_name, value in book_payload.metadata.items():
+        if metadata_name.startswith("Illustration_"):
+            await TitleMetadata.objects.create(
+                title=title.ident,
+                name=metadata_name,
+                bin_value=base64.standard_b64decode(value),
+                kind=KIND_ILLUSTRATION,
+            )
+        else:
+            await TitleMetadata.objects.create(
+                title=title.ident,
+                name=metadata_name,
+                value=value,
+                kind=KIND_TEXT,
+            )
 
     for tag_name in book_payload.metadata["Tags"].split(";"):
         book_tag = await BookTag.objects.get_or_create(name=tag_name)
         await book.tags.add(book_tag)
-        await title.tags.add(book_tag)
+        if not tag_name.startswith("_") or "_category" in tag_name:
+            await title.tags.add(book_tag)
 
     for lang_code in book_payload.metadata["Language"].split(","):
-        lang_dict = get_language_details(lang_code)
+        lang_dict = find_language_names(lang_code)
```

Comment:
Using the type of a variable in its name is usually not a good idea. It's generally the sign its name if not clear enough. Here it's even wrong as this is a tuple.
Since this function returns a tuple and you'll use both values, create two variables: 

```py
english_name, native_name = find_language_names(lang_code)
```